### PR TITLE
Disable NScale resolution scaling for Android

### PR DESCRIPTION
**Pull request by ltrung**
**Issue #1736:**
Android devices' H.264 video encoding might have issue when the nscale video uplink bandwidth policy tried to scale down the resolution when the number of videos on the call increases. There is a Chromium bug opened about this ([link](https://bugs.chromium.org/p/chromium/issues/detail?id=1203613&q=android%20scaleResolutionDownBy&can=2)).
This will cause frozen or blank video on remote clients.

**Description of changes:**
Disable resolution scaling for Android for now. 

**Testing:**

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*
- Join the meeting demo with at least 4,8 ,12,  18, 25 videos and one from Android user.
- Make sure that the Android user is not the active speaker.
- Verify that the resolution of the Android user does not change and there is no frozen or blank video on the other side.

**Checklist:**

1. Have you successfully run `npm run build:release` locally? Yes

2. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved? Yes adding one API to check whether we should disable resolution scaling.

3. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved? No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/browserbehavior/ExtendedBrowserBehavior.ts [hunk 1]
```diff
@@ -18,4 +18,5 @@ export default interface ExtendedBrowserBehavior extends BrowserBehavior {
   requiresContextRecreationForAudioWorklet(): boolean;
   isSimulcastSupported(): boolean;
   supportsBackgroundFilter(): boolean;
+  supportsResolutionScaling(): boolean;
```
Comment (hensmi-amazon) [inline src/browserbehavior/ExtendedBrowserBehavior.ts:21]: feels a bit unnecessarily unspecific.  Also 'support' feels like the wrong word.  `shouldDisableScaleDownBy`? or something?

Comment (ltrung) [reply src/browserbehavior/ExtendedBrowserBehavior.ts]: change to `disableScaleResolutionDownBy`

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,13 +20,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Take into account portrait video dimension when calculating resolution scaling for nscale video bandwidth uplink 
+  policy.
   
 ### Changed
 
 - Ignore `enableUnifiedPlanForChromiumBasedBrowsers` value (i.e. treat as always equaling the current default value of `true`) in `MeetingSesstionConfiguration`.  Chrome is [in the processing](https://groups.google.com/g/discuss-webrtc/c/UBtZfawdIAA/m/m-4wnVHXBgAJ) of deprecating and removing Plan-B which would cause breakage in applications still trying to use it.  This will have no effect on SDK behavior` and has been the default since 1.17.0.
 - Change `appVersionName` and `appVersionCode` fields to `appName` and `appVersion` respectively.
 - Update similar log messages in `DefaultMessagingSession` and `DefaultSignalingClient`.
 - Change the serverless demo's `deploy.js` script to rebuild the demo on each run. This should remove a manual step of rebuilding the demo.
+- Disable nscale resolution scaling for Android device due to Android H.264 encoding.
```
Comment (devalevenkatesh) [inline CHANGELOG.md:32]: Should we expand on the issue a bit here, or tag any existing issues for clarity?

File: src/videouplinkbandwidthpolicy/NScaleVideoUplinkBandwidthPolicy.ts [hunk 3]
```diff
@@ -203,7 +209,12 @@ export default class NScaleVideoUplinkBandwidthPolicy implements VideoUplinkBand
             NScaleVideoUplinkBandwidthPolicy.targetHeightArray.length - 1
           )
         ];
-      scale = Math.max(setting.height / targetHeight, 1);
+      scale = Math.max(Math.min(setting.height, setting.width) / targetHeight, 1);
```
Comment (devalevenkatesh) [inline src/videouplinkbandwidthpolicy/NScaleVideoUplinkBandwidthPolicy.ts:212]: [For understanding]: Is this the portrait condition handling?

Comment (ltrung) [reply src/videouplinkbandwidthpolicy/NScaleVideoUplinkBandwidthPolicy.ts:212]: Yep

File: CHANGELOG.md (hunk 2 above)
Comment (devalevenkatesh) [reply CHANGELOG.md:32]: Added info in PR description and we have not been adding issue tags in changelogs.

**Review by devalevenkatesh — APPROVED** [review]

**Pull request merged**